### ISSUE TITLE
IB: generated code doesn't compile

### ISSUE THREAD

**Issue by Megatron1000** [SwiftGen/SwiftGen#515]
Using the new 6.0.0 release when using the IB templates to build a macOS project the files output by the segues template and scenes template don't compile. 

Command used it this:
```sh
swiftgen ib "${PROJECT_DIR}/XcodeProject/Base.lproj/Main.storyboard" -output "${PROJECT_DIR}/XcodeProject/SwiftGenGenerated/StoryboardScenes.swift" -t scenes-swift4
swiftgen ib "${PROJECT_DIR}/XcodeProject/Base.lproj/Main.storyboard" -output "${PROJECT_DIR}/XcodeProject/SwiftGenGenerated/StoryboardSegues.swift" -t segues-swift4
```

**Comment by Megatron1000** [comment]
There appears to be two issues. 

- Both templates need `import Cocoa` added under `import Foundation`

- Segues need to change  `guard let identifier = segue.identifier?.rawValue else { return nil }` into `guard let identifier = segue.identifier else { return nil }`

**Comment by AliSoftware** [comment]
Thanks for reporting the issue @Megatron1000 !

What surprises me is that our CI didn't catch it. If I run out task responsible for checking that the generated output of `scenes-swift4` ib template compiles on macOS, it indeeds pass without any compilation error… I wonder what's different then

```
$ rake compile:output
[…]
Compiling Tests/Fixtures/Generated/IB-macOS/scenes-swift4-context-all.swift…
 xcrun --toolchain com.apple.dt.toolchain.XcodeDefault -sdk macosx swiftc -swift-version 4 -typecheck -target x86_64-apple-macosx10.13 -I "Tests/Fixtures/CompilationEnvironment/swift4/macosx"  -module-name SwiftGen Tests/Fixtures/CompilationEnvironment/Definitions.swift Tests/Fixtures/Generated/IB-macOS/scenes-swift4-context-all.swift
[…]
```

**Comment by AliSoftware** [comment]
@Megatron1000 What version of Xcode and Swift are you using? Maybe it's a slight difference between, say, Swift 4.0 in Xcode 9.x and Swift 4.2 in Xcode 10, or similar?

**Comment by Megatron1000** [comment]
Well, now might be a good time to mention this is Xcode 10, Mojave and Swift 4.2. 
That looks CI looks like it's Swift 4.

**Comment by AliSoftware** [comment]
For information, I ran that `rake output:compile` test I mentioned above on my mac, with `Xcode-select` pointing to Xcode 10.0 (still on macOS 10.13.6 and not Mojave here, but shouldn't affect it?). And when we're compiling from the `swiftc` CLI with `-swift-version 4` from the Xcode 10 toolchain, I think that uses Swift 4.2, right?

**Comment by Megatron1000** [comment]
FYI just switched my project's Swift version to Swift 4.0 and the `guard let identifier = segue.identifier?.rawValue else { return nil }` problem went away.

**Comment by Megatron1000** [comment]
So looks to me like that's a language problem.

**Comment by Megatron1000** [comment]
The import one I'm not sure about. However, the build error is that it doesn't recognise `NSStoryboard` and that is part of `AppKit`. The only import in my file is `Foundation`, so seems to make sense that it wouldn't compile.

**Comment by Megatron1000** [comment]
Haven't had any recent experience with `xcode build` so couldn't tell you what that setting's doing, I'm afraid.

**Comment by AliSoftware** [comment]
Ah interesting!

So we should make our CI build both in 4.0 and 4.2 mode then, to detect such failures; not sure what's the swiftc flag for that, is it just `-swift-version 4.2` 🧐

Still doesn't explain why, even in Swift 4.0 mode that is tested by our CI, the missing import doesn't seem to bother it (while it's required, as you said)

**Comment by AliSoftware** [comment]
Thanks for the detective work btw 👍

**Comment by Megatron1000** [comment]
One other potential reason I can think that the `import` might behave differently on my machine compared to yours is I'm building using the `New Build System`. I wonder if that's the same one that's invoked by your command. I tried switching to the legacy build system and it still had the same problem, so probably not, but thought it might be worth mentioning.

**Comment by rudedogg** [comment]
Hmm, I'm seeing the same issue. I don't have any issue with the `import`(s) on my end, AppKit/Foundation are listed which seems to work. The only error I'm getting from Xcode is on the `guard let identifier = segue.identifier?.rawValue else { return nil }`

**Comment by djbe** [comment]
@Megatron1000 could it be that there's not a single scene in your storyboard file that is a basic type? And with that, I mean, have you set a custom class for every scene?

**Comment by AliSoftware** [comment]
@djbe I've made a PR so that we now test both Swift 4.0 and 4.2.

That PR is expected to fail compile IB-macOS files in Swift 4.2 mode on CI — At least it properly fails on my machine, even with our unit tests. Once I confirm it's red, we should push a commit on that same PR to fix the templates for 4.2 to fix the original issue.

**Comment by Megatron1000** [comment]
> @Megatron1000 could it be that there's not a single scene in your storyboard file that is a basic type? And with that, I mean, have you set a custom class for every scene?

@djbe You're right! That seems to be the cause of the missing imports. I had one `NSWindowController` that wasn't one of my own custom subclasses in the storyboard. As soon as I gave that `NSWindowController` a `Storyboard ID`, so it wouldn't be ignored by `SwiftGen`, the AppKit import was added to the generated files.

**Comment by lordcodes** [comment]
As a note, the same import issue exists for iOS. The 'import UIKit' is missing from the stencil, so the generated storyboards files don't compile. You need to add 'import UIKit' to the scenes-swift4.stencil file in order to use SwiftGen 6.0.0.

**Comment by AliSoftware** [comment]
@lordcodes Thanks for the report.

This should already have been fixed in `develop` PR #519 (if you have the occasion to test develop to confirm). We'll do an official bugfix release soon :wink:
